lxd/storage/drivers/driver_zfs_volumes.go

```diff
@@ -2835,14 +2844,25 @@ func (d *zfs) MountVolumeSnapshot(snapVol Volume, op *operations.Operation) erro
 					d.logger.Debug("Regenerating filesystem UUID", logger.Ctx{"dev": volPath, "fs": tmpVolFsType})
 					err = regenerateFilesystemUUID(mountVol.ConfigBlockFilesystem(), volPath)
 					if err != nil {
-						return err
+						return nil, err
+					}
+				}
+			} else {
+				// ext4 will replay the journal if the filesystem is dirty.
```

Comment:
This comment and logic is repeated twice in this PR. It feels like we could do with a driver level function called something like `readOnlyMountOpts(fsType string)` where you can encapsulate that logic and comment once.

Reply:
Also, if you specify the same option twice, does it cause any issues? I'm wondering if we need to have the de-dupe logic at all.